### PR TITLE
Expose author information in add-on search and detail API

### LINKED ISSUE #3022: Return add-on author information in detail API
The detail API needs to return author(s) information for an add-on. The detail page needs to show the full name (`name` property, which uses `display_name` if available, falling back to `username`) and the url to its profile.

We might need to tweak the search index if we also want to return this in the search API, because the index only contains authors names right now, not their usernames, so it can not return the links.

Supports mozilla/addons/issues/151

### PR DESCRIPTION
**Pull request by diox**
Fix #3022

### CODE REVIEW

File: src/olympia/addons/indexers.py [hunk 1]
```diff
@@ -36,6 +36,18 @@ def get_mapping(cls):
                     'app': {'type': 'byte'},
                     'appversion': {'properties': {app.id: appver
                                                   for app in amo.APP_USAGE}},
+                    'author': {
+                        'type': 'object',
+                        'properties': {
+                            'id': {'type': 'long', 'index': 'no'},
+                            'name': {'type': 'string', },
```
Comment (EnTeQuAk) [inline src/olympia/addons/indexers.py]: I'm not a fan of the ending comma and whitespace here inside the dict.

File: src/olympia/addons/indexers.py [hunk 2]
```diff
@@ -36,6 +36,18 @@ def get_mapping(cls):
...
+                            'username': {'type': 'string', 'index': 'no', },
```
Comment (EnTeQuAk) [inline src/olympia/addons/indexers.py]: Same here

File: src/olympia/addons/indexers.py [hunk 3]
```diff
@@ -176,6 +188,12 @@ def extract_document(cls, obj):
                 'min': min_, 'min_human': min_human,
                 'max': max_, 'max_human': max_human,
             }
+        data['author'] = [{'name': a.name, 'id': a.id, 'username': a.username}
```
Comment (EnTeQuAk) [inline src/olympia/addons/indexers.py]: I'm not sure I like the naming. `author` implies only one object, yet it's a list of objects. Can't we just name it `listed_authors` too? Or do you plan on just removing `authors` and switch over?

**Comment by EnTeQuAk** [comment]
r+wc

File: src/olympia/addons/indexers.py (hunk 3 above)
Comment (diox) [inline src/olympia/addons/indexers.py]: mmm I replied to this but it's not showing, maybe I forgot to hit the button. Anyway yes, `listed_authors` is better, I'll change it.

**Pull request merged**